core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -144,23 +144,40 @@ private TableDataMap getTableDataMap(String dataMapName,
    * Clear the datamap/datamaps of a table from memory
    * @param identifier Table identifier
    */
-  public void clearDataMap(AbsoluteTableIdentifier identifier) {
+  public void clearDataMaps(AbsoluteTableIdentifier identifier) {
     List<TableDataMap> tableDataMaps = allDataMaps.get(identifier.uniqueName());
     segmentRefreshMap.remove(identifier.uniqueName());
     if (tableDataMaps != null) {
-      int i = 0;
       for (TableDataMap tableDataMap: tableDataMaps) {
         if (tableDataMap != null) {
           tableDataMap.clear();
           break;
         }
-        i++;
       }
       allDataMaps.remove(identifier.uniqueName());
     }
   }
 
   /**
+   * Clear the datamap/datamaps of a table from memory
+   * @param identifier Table identifier
+   */
+  public void clearDataMap(AbsoluteTableIdentifier identifier, String dataMapName) {
```

Comment:
contains bug in removing element

Reply:
what bug?

Reply:
sorry, didn't notice the `break` before. That's fine~